packages/near-fast-auth-signer/src/components/SignMultichain/utils.ts

```diff
@@ -167,31 +205,41 @@ export const shortenAddress = (address: string): string => {
 };
 
 export const multichainSignAndSend = async ({
-  domain,
-  asset,
-  to,
-  value,
+  signMultichainRequest,
   feeProperties
 }: {
-  domain: string;
-  asset: Chain;
-  to: string;
-  value: string;
+  signMultichainRequest: EvmSendMultichainMessage | BTCSendMultichainMessage;
   feeProperties: TransactionFeeProperties;
 }) => {
   const accountId = window.fastAuthController.getAccountId();
   const keypair = await window.fastAuthController.getKey(accountId);
-  const derivedPath = `,${asset},${domain}`;
+  const chainDetails = {
+    chain:   signMultichainRequest.chain,
+    chainId: (signMultichainRequest as EvmSendMultichainMessage).chainId,
```

Comment:
Instead of casting, can we do it with type guards?

I know it will be more code, but add some extra safety and readability to the code.